src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -197,77 +210,84 @@ public synchronized boolean open() throws CouchbaseLiteException {
 
         // Incompatible version changes increment the hundreds' place:
         if (dbVersion >= 200) {
-            Log.e(TAG, "Database: Database version (%d) is newer than I know how to work with",
-                    dbVersion);
             storageEngine.close();
-            return false;
+            String message = "Database version " + dbVersion +
+                    " is newer than I know how to work with";
+            Log.e(TAG, message, dbVersion);
+            throw new CouchbaseLiteException(message, Status.NOT_ACCEPTABLE);
         }
 
-        boolean isNew = (dbVersion == 0);
-        boolean isSuccessful = false;
-
         // BEGIN TRANSACTION
+        boolean isSuccessful = false;
         if (!beginTransaction()) {
             storageEngine.close();
-            return false;
+            String message = "Cannot begin transaction";
+            Log.e(TAG, message);
+            throw new CouchbaseLiteException(message, Status.DB_ERROR);
         }
 
         try {
+            boolean isNew = (dbVersion == 0);
             if (dbVersion < 17) {
                 // First-time initialization:
                 // (Note: Declaring revs.sequence as AUTOINCREMENT means the values will always be
                 // monotonically increasing, never reused. See <http://www.sqlite.org/autoinc.html>)
-
                 if (!isNew) {
-                    Log.w(TAG, "Database version (%d) is older than I know how to work with",
-                            dbVersion);
-                    storageEngine.close();
-                    return false;
+                    String message = "Database version " + dbVersion +
+                            " is older than I know how to work with";
+                    Log.e(TAG, message, dbVersion);
```

Comment:
In format, no %d. But method is given dbVersion as third parameter. I am not sure what kind of problem this might cause.